spec/features/user_interacts_with_the_queue_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'spec_helper'
+
+feature "User interacts with the queue" do 
+  scenario "user adds and reorders videos in the queue" do 
+    category = Fabricate(:category)
+    video1   = Fabricate(:video, category: category)
+    video2   = Fabricate(:video, category: category)
+    video3   = Fabricate(:video, category: category)
+
+    sign_in
+    add_video_to_queue(video1)
+    expect_video_to_be_in_queue(video1)
+
+    visit video_path(video1)
+    expect_link_to_be_hidden("+ My Queue")
+
+    add_video_to_queue(video2)
+    add_video_to_queue(video3)
+
+    set_video_position(video1, 3)
+    set_video_position(video2, 1)
+    set_video_position(video3, 2)
+
+    update_queue
+
+    expect_video_position(video1, "3")
+    expect_video_position(video2, "1")
+    expect_video_position(video3, "2")
+  end
+
+  def add_video_to_queue(video)
+    visit home_path
+    find("a[href='/videos/#{video.id}']").click
+    click_link "+ My Queue"
+  end
+
+  def expect_video_to_be_in_queue(video)
+    page.should have_content(video.title)
+  end
+
+  def expect_link_to_be_hidden(link)
+    page.should_not have_content(link)
```

Comment:
Actually you are using the `should` version, so that would be instead of using `should_not`. Also, the `expect` syntax is preferred over the `should` syntax now.